src/uri.cr

```diff
@@ -128,7 +128,7 @@ class URI
   def full_path
     String.build do |str|
       str << (@path.try { |p| !p.empty? } ? @path : "/")
-      str << "?" << @query if @query
+      str << "?" << @query if @query && !@query.try(&.empty?)
```

Comment:
Better: `if (query = @query) && !query.empty?`

Reply:
I like that 👍